src/renderers/materialized/materialized.text.control.tsx

```diff
@@ -1,17 +1,34 @@
 import { JSX } from '../JSX';
 import { withIncreasedRank } from '../../core/testers';
 import { Control, ControlProps, ControlState } from '../controls/Control';
+import { resolveSchema } from '../../path.util';
+import { ControlElement } from '../../models/uischema';
 import { mapStateToControlProps, registerStartupRenderer } from '../renderer.util';
 import { textControlTester } from '../controls/text.control';
 import { connect, Event } from '../../common/binding';
-
+declare let $;
 export class MaterializedTextControl extends Control<ControlProps, ControlState> {
 
+  componentDidMount() {
+    const { id, uischema, schema } = this.props;
+    const controlElement = uischema as ControlElement;
+    const maxLength = resolveSchema(schema, controlElement.scope.$ref).maxLength;
+    if (uischema.options && uischema.options.trim && maxLength !== undefined) {
+      const fontSize = parseFloat($('[id="' + id + '"]').css('font-size'));
+      $('[id="' + id + '"]').css('width', (maxLength * 15) * (fontSize / 14.5) + 'px');
```

Comment:
Can you explain the constants?

Reply:
First part: W is the widest letter of the latin alphabet. It takes up 15px, so we multiply by the maximum number of characters (maxLength).

Second part: The initial font size is 14.5px, so increasing the (calculated) font-size property means we have to multiply the first part by the ratio of which the font size has increased compared to the initial 14.5px